test/replcompletions.jl

```diff
@@ -494,6 +493,29 @@ c, r, res = test_scomplete(s)
         ENV["PATH"] = oldpath
     end
 
+    # Make sure completion results are unique in case things are in the env path twice.
+    let
+        # Create /tmp/repl-completion and /tmp/repl-completion-subdir/repl-completion
+        file0 = joinpath(tempdir(), "repl-completion")
+        dir = joinpath(tempdir(), "repl-completion-subdir")
+        mkdir(dir)
+        file1 = joinpath(dir, "repl-completion")
+        touch(file0)
+        touch(file1)
+
+        oldpath = ENV["PATH"]
+        ENV["PATH"] = string(tempdir(), ":", dir)
```

Comment:
better to use withenv here. are you sure this needs to be unix only, or would it work on windows too (with the path separator as `;` there)? aside from the tests in this unix-only block that assume tempdir starts with `/tmp` i would hope more of them should be portable and tested everywhere.